src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > table.length * LOAD_FACTOR) {
+            resize();
+        }
```

Comment:
move if to resize() method, because its part of resize() method logic and never gonna be used without if check